### ISSUE TITLE
New function create_webservice

### ISSUE THREAD

**Issue by pycaret** [pycaret/pycaret#492]
Create a new function called create_webservice and use FastAPI to deploy web service on localhost. 

FastAPI will be optional dependency of PyCaret. 

## Who can work on this feature?

**Comment by g0lemXIV** [comment]
I could start doing it at the beginning of September (I'm going on vacation next week), now I'm creating my own API, base on FastApi. We could talk about it more, and establish endpoints and schemes for that, in September.

**Comment by Yard1** [comment]
@g0lemXIV What's the progress on this?

**Comment by g0lemXIV** [comment]
@Yard1 I've just back from the holiday. I will start implementing web service this week. 

If I understand correctly, the `create_webservice` function in the basic version should take model, model_name, port number (optional), and path (optional). If the user trigger the function, fast API initializes REST API with POST endpoint and return prediction in the body.

In the next version, I can add authentication, counter, and database for prediction (optional) which can be useful for monitoring models.

I'm not sure how model ensembling works in pycaret, and how to implement it in rest API. But we could think about it in the next release.

**Comment by pycaret** [comment]
@g0lemXIV Your understanding is 100% correct. plus obviously incorporate the dependencies in `requirements.txt` as well. We will keep `FastAPI` optional / commented in `requirements.txt` and in the docstring of `create_webservice` we can put a warning and an exception check on top. See exception checking for `interpret_model` where I have done a similar check to detect `shap` before executing the code.

If you can think of a better name then `create_webservice` I will be happy to hear that. If not, let's proceed with `create_webservice` for now.

**Comment by pycaret** [comment]
@g0lemXIV All okay?

**Comment by g0lemXIV** [comment]
@pycaret, yes sorry that I didn't respond. I'm creating it right now, it should be implemented in all main files, yes? (classification, regression, etc.)

**Comment by g0lemXIV** [comment]
I missed it. I think, will pull request the first version on Sunday/Saturday. Should I add Pydantic to check the input data format?

**Comment by g0lemXIV** [comment]
@pycaret  @Yard1 Tomorrow I will pull request the version with the web service. I did all the functionality except tests, docs, and requirements completion. Basic structure looks this:
`create_webservice(model, model_name, port=8080, api_key=True,
                      use_pydatic=True, pydantic_payload=None)`

api_key is responsible for generating a basic security level;
pydatic is responsible for the proper structure on input and output of the api.

**Comment by pycaret** [comment]
@g0lemXIV Great thanks. Will, you also include dependencies in `requirements.txt` (though should be commented out just like other optional dependencies)

**Comment by ngupta23** [comment]
@Yard1 @moezali1 
This seems to be old, but is this functionality different from the newly added `create_api`?

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
